File: CPP/human.cpp
// import library
#include <iostream>
#include <string>
#include <list>
#include <bits/stdc++.h>

// standard namespace
using namespace std;

// class human
class human
{
private:
    string nik, nama, kelamin;

public:
    // constructor
    human()
    {
        this->nik = "";
        this->kelamin = "";
        this->nama = "";
    }
    // setter dan getter
    string getnama()
    {
        return this->nama;
    }
    string getnik()
    {
        return this->nik;
    }
    string getkelamin()
    {
        return this->kelamin;
    }

    void setnama(string nama)
    {
        this->nama = nama;
    }
    void setnik(string nik)
    {
        this->nik = nik;
    }
    void setkelamin(string kelamin)
    {
        this->kelamin = kelamin;
    }
    // destructor
    ~human()
    {
    }
};

// class sivitas sebagai child dari human
class sivitas : public human
{

private:
    string universitas, email;

public:
    // constructor
    sivitas()
    {
        this->email = "";
        this->universitas = "";
    }
    // setter dan getter
    void setuniversitas(string universitas)
    {
        this->universitas = universitas;
    }
    void setemail(string email)
    {
        this->email = email;
    }
    string getuniversitas()
    {
        return this->universitas;
    }
    string getemail()
    {
        return this->email;
    }
    // destructor
    ~sivitas()
    {
    }
};

// class mahasiswa sebagai child dari sivitas
class mahasiswa : public sivitas
{
private:
    string nim, fakultas, prodi;

public:
    // constructor
    mahasiswa()
    {
        this->nim = "";
        this->fakultas = "";
        this->prodi = "";
    }

    // setter dan getter
    void setnim(string nim)
    {
        this->nim = nim;
    }
    void setfakultas(string fakultas)
    {
        this->fakultas = fakultas;
    }
    void setprodi(string prodi)
    {
        this->prodi = prodi;
    }
    string getnim()
    {
        return this->nim;
    }
    string getfakultas()
    {
        return this->fakultas;
    }
    string getprodi()
    {
        return this->prodi;
    }
    // destructor
    ~mahasiswa()
    {
    }
};

File: PHP/human.php
<?php
//class human 
class human
{
    //atribut private
    private $nik;
    private $nama;
    private $kelamin;

    //constructor
    public function __construct()
    {
        $this->nik = "";
        $this->nama = "";
        $this->kelamin = "";
    }

    //setter dan getter
    public function setnik($nik)
    {
        $this->nik = $nik;
    }
    public function setnama($nama)
    {
        $this->nama = $nama;
    }
    public function setkelamin($kelamin)
    {
        $this->kelamin = $kelamin;
    }
    public function getnik()
    {
        return $this->nik;
    }
    public function getnama()
    {
        return $this->nama;
    }
    public function getkelamin()
    {
        return $this->kelamin;
    }
    //destructor
    function __destruct()
    {
    }
}

//class sivitas child dari human
class sivitas extends human
{
    //atribut private
    private $universitas;
    private $email;
    //constructor

    public function __construct()
    {
        $this->universitas = "";
        $this->email = "";
    }
    //setter dan getter

    public function setuniversitas($universitas)
    {
        $this->universitas = $universitas;
    }
    public function setemail($email)
    {
        $this->email = $email;
    }
    public function getuniversitas()
    {
        return $this->universitas;
    }
    public function getemail()
    {
        return $this->email;
    }
    //destructor
    function __destruct()
    {
    }
}

class mahasiswa extends sivitas
{
    //atribut private
    private $nim;
    private $fakultas;
    private $prodi;

    //constructor
    public function __construct()
    {
        $this->nim = "";
        $this->fakultas = "";
        $this->prodi = "";
    }
    //setter dan getter

    public function setfakultas($fakultas)
    {
        $this->fakultas = $fakultas;
    }
    public function setnim($nim)
    {
        $this->nim = $nim;
    }
    public function setprodi($prodi)
    {
        $this->prodi = $prodi;
    }
    public function getfakultas()
    {
        return $this->fakultas;
    }
    public function getnim()
    {
        return $this->nim;
    }
    public function getprodi()
    {
        return $this->prodi;
    }
    //destructor
    function __destruct()
    {
    }
}
